### DISCUSSION TITLE
⚠️Could the new INP (Interaction to next paint) metric from Google Core Web Vitals be a problem for Nuxt?

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#25960] [General]
We have a relatively new website built with Nuxt 3 and Storyblok as CMS. 
We prerender almost all sites and we have really good real life user scores in all Google pagespeed metrics.
But the new metric INP is 287ms on mobile, which is not optimal (on desktop its 98 so its okay). 

![image](https://github.com/nuxt/nuxt/assets/93664840/5341f496-780a-4044-b4e1-30768c9b3a5b)

I've also looked at some other nuxt sites and https://nuxt.com/ for example has an INP of 604ms on mobile! (https://pagespeed.web.dev/analysis/https-nuxt-com/5nu8bd42xc?form_factor=mobile)
Other cases with similar scores as us:
OpenAI: https://pagespeed.web.dev/analysis/https-openai-com/nqi508yurn?form_factor=mobile
Ecosia: https://pagespeed.web.dev/analysis/https-www-ecosia-org/mk6jl9e1pc?form_factor=mobile
Tiktok B2B: https://pagespeed.web.dev/analysis/https-www-tiktok-com-business-es/ol9rsuh9c8?form_factor=mobile

I've read that because of the high load on the main thread, frameworks like Nuxt or React have a much higher INP than other websites (e.g. our old wordpress website is at 172ms right now, which is good enough for Google). (@danielroe maybe you have more insight?)

I know that we have quite a lot of features and interactions on our website and the sites themselves are sometimes quite long, so that could also be the cause, but **i wanted to ask if anyone gets similar results or knows how to reduce it on weaker devices**.

Thanks!

**Comment by HavishNetla** [comment]
Agreed!

**Comment by VoytenkoVitaliyOlegovich** [comment]
Agreed!

**Comment by ghost** [comment]
Update: INP is now live and all our Nuxt pages got put in "should be optimized" in the search console :(
![image](https://github.com/nuxt/nuxt/assets/93664840/bed65c9f-d87c-4ada-9975-9d9542a9ab9a)

**Comment by ghost** [comment]
@ all who have the same problem:
 the reasons for a higher INP is very varied. I think I have a very specific problem (among other smaller problems) with the use of a headless CMS (Storyblok) and prerendering (Very high INP on second navigating to the same site). You can see it here: https://github.com/nuxt/nuxt/issues/26271

An extremely usefull tool is the Web Vitals Chrome plugin 💪. It logs in the console every interaction speed: https://chromewebstore.google.com/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma

If you use Nuxt without special settings most of the time there is something to be improved on, on how you call data and how you show it to users. Maybe use `useAsyncData` without await and show a pending/loading state (e.g. skeletons) to the user. Also there is `NuxtLoadingIndicator` if you're pages take quite longer when navigating, I don't know if it makes a difference, but I think its good UX to show the user something is loading.

Please share how you host your sites and what problems you face :) @HavishNetla @VoytenkoVitaliyOlegovich

**Comment by InstantlyMoist** [comment]
Currently facing the same issue here.. The extension you mentioned @LukasHechenbergerID works fine, but unfortunately isn't representative of what Google thinks.

Even with the slow 4g throttling it's always 10-15ms before interaction. Pagespeed tells us it's 235ms. Any ideas on how to find out what could be causing this? The stats are fine on desktop too.

**Reply by ghost** [reply]
10-15ms is very good, so the interactions you measured shouldn't be the issue. 

INP is very dependend on the highest interaction time a user has on your site (more here: https://web.dev/articles/inp). So it could be that only one interaction is the cause of the problem (e.g. opening your mobile menu, a navigating/payload bug like mine, a carousel, a button,...). 
It could also be dependend on what tech you are using with Nuxt (CMS?, Images?), do you have big requests?

Feel free to share more infos about your situation and maybe a link, then I could look at it myself if you want :)

**Reply by InstantlyMoist** [reply]
@LukasHechenbergerID the site i'm working on is https://minecraftiplist.com

I'm honestly unable to find anything that might cause it, not in the mobile design nor the desktop design.

**Reply by ghost** [reply]
As I see it your main problem isn't INP:
![image](https://github.com/nuxt/nuxt/assets/93664840/c8b399dd-e68d-4629-b57d-8f191d460204)
Try to lazy load more with
`const { data, pending, error, refresh } = await useFetch(...);`
You can use the pending state.
And add loaders when searching or selecting a category/tag.

But I couldn't find any problems with the INP either so thats strange... (i just noticed that when you click on a tag it doesn't show anything new for a second and then switches the servers, maybe this is a reason... )
